dashboard/api.py

```diff
@@ -207,27 +209,29 @@ def _add_run(run, mmtrack_, status):
     run_statuses.remove(run_status)
 
     if run_status.status == CourseRunStatus.NOT_ENROLLED:
-        next_run = course.get_next_run()
+        next_run = course.get_first_unexpired_run()
         if next_run is not None:
             _add_run(next_run, mmtrack, CourseStatus.OFFERED)
     elif run_status.status == CourseRunStatus.NOT_PASSED:
-        next_run = course.get_next_run()
+        _add_run(run_status.course_run, mmtrack, CourseRunStatus.NOT_PASSED)
+        next_run = course.get_first_unexpired_run()
         if next_run is not None:
             _add_run(next_run, mmtrack, CourseStatus.OFFERED)
-        if next_run is None or run_status.course_run.pk != next_run.pk:
-            _add_run(run_status.course_run, mmtrack, CourseStatus.NOT_PASSED)
+    elif run_status.status == CourseRunStatus.MISSED_DEADLINE:
+        _add_run(run_status.course_run, mmtrack, CourseRunStatus.MISSED_DEADLINE)
+        future_run = course.get_first_future_run()
```

Comment:
Why are we using `get_first_future_run` instead of `get_first_unexpired_run` here?